src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs

```diff
@@ -568,22 +570,44 @@ public Task<T> InvokeDaasExtension<T>(string subscriptionId, string resourceGrou
 
         private Task<R> HttpGet<R>(string path, string queryString = "", string apiVersion = GeoMasterConstants.August2016Version, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return PerformHttpRequest<R>(HttpMethod.Get, path, queryString, null, apiVersion, cancellationToken);
+            //return PerformHttpRequest<R>(HttpMethod.Get, path, queryString, null, apiVersion, cancellationToken);
+            return PerformHttpRequestWithRetry<R>(HttpMethod.Get, path, queryString, null, apiVersion, cancellationToken);
         }
 
         private Task<R> HttpPost<R, T>(string path, T content = default(T), string queryString = "", string apiVersion = GeoMasterConstants.August2016Version, CancellationToken cancellationToken = default(CancellationToken))
         {
             var body = JsonConvert.SerializeObject(content);
-            return PerformHttpRequest<R>(HttpMethod.Post, path, queryString, body, apiVersion, cancellationToken);
+            //return PerformHttpRequest<R>(HttpMethod.Post, path, queryString, body, apiVersion, cancellationToken);
+            return PerformHttpRequestWithRetry<R>(HttpMethod.Post, path, queryString, body, apiVersion, cancellationToken);
         }
 
+        private Task<R> PerformHttpRequestWithRetry<R>(HttpMethod method, string path, string queryString, string content, string apiVersion, CancellationToken cancellationToken)
+        {
+            var param = new GeoMasterRetryParam()
+            {
+                HttpMethod = method,
+                Path = path,
+                QueryString = queryString,
+                Content = content,
+                ApiVersion = apiVersion,
+                CancellationToken = cancellationToken
+            };
+            
+            var task = RetryHelper.RetryAsync<R>(PerformHttpRequest<R>,param, "GeoMasterDataProvider", RequestId, _configuration.MaxRetryCount, _configuration.RetryDelayInSeconds * 1000);
```

Comment:
nitpick : spacing between arguments. Just format the whole document :)

Reply:
Yes, will do